File: VB/WebApplication1/Service1.svc.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Linq
Imports System.Runtime.Serialization
Imports System.ServiceModel
Imports System.Text
Imports System.Threading

Namespace WebApplication1
    <KnownType(GetType(Item))> _
    Public Class Service1
        Implements IService1
        Public Function GetData(ByVal skipCount As Integer, ByVal takeCount As Integer) As IEnumerable Implements IService1.GetData
            Dim endIndex As Integer = skipCount + takeCount
            Debug.WriteLine(String.Format("skipCount={0}, takeCount={1}", skipCount, takeCount))
            Dim items As New List(Of Item)()
            For i As Integer = skipCount To endIndex - 1
                items.Add(New Item With {.Id = i, .Name = "Name" + i.ToString()})
            Next i
            Return items
        End Function
        Public Function Count() As Integer Implements IService1.Count
            Return 1000
        End Function
    End Class
    Public Class Item
        Implements IItem

        <DataMember> _
        Public Property Id() As Integer Implements IItem.Id
        <DataMember> _
        Public Property Name() As String Implements IItem.Name
    End Class
End Namespace

File: CS/WebApplication1/Service1.svc.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;
using System.Threading;

namespace WebApplication1 {
    // NOTE: You can use the "Rename" command on the "Refactor" menu to change the class name "Service1" in code, svc and config file together.
    // NOTE: In order to launch WCF Test Client for testing this service, please select Service1.svc or Service1.svc.cs at the Solution Explorer and start debugging.
    [KnownType(typeof(Item))]
    public class Service1: IService1 {
        public IEnumerable GetData(int skipCount, int takeCount) {
            int endIndex = skipCount + takeCount;
            Debug.WriteLine(string.Format("skipCount={0}, takeCount={1}", skipCount, takeCount));
            List<Item> items = new List<Item>();
            for(int i = skipCount; i < endIndex; i++)
                items.Add(new Item { Id = i, Name = "Name" + i });
            return items;
        }
        public int Count() {
            return 1000;
        }
    }
    public class Item: IItem {
        [DataMember]
        public int Id { get; set; }
        [DataMember]
        public string Name { get; set; }
    }
}